pkg/astvalidation/reference/main.go

```diff
@@ -0,0 +1,461 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+//go:generate ./gen.sh
+
+func main() {
+	currDir, _ := os.Getwd()
+	println(currDir)
+
+	workingDir := inputDir
+	if !strings.Contains(currDir, "reference") {
+		workingDir = filepath.Join(currDir, "pkg/astvalidation/reference/__tests__")
+	}
+
+	dir, err := ioutil.ReadDir(workingDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	replacementsPath := workingDir + "/../replacements.yml"
+	replacementContent, _ := ioutil.ReadFile(replacementsPath)
+
+	var replacements []Replacement
+	if err := yaml.Unmarshal(replacementContent, &replacements); err != nil {
+		log.Fatal(err)
+	}
+
+	for _, fileInfo := range dir {
+		processFile(workingDir, fileInfo.Name(), replacements)
+	}
+}
+
+const (
+	inputDir = "./__tests__"
+	outDir   = "./testsgo"
+
+	header = `
+package testsgo
+
+import (
+  "testing"
+)
+`
+)
+
+var (
+	// jsArrowFunction - represents a function passed to mocha describe and it
+	jsArrowFunction = ", () => {"
+
+	// convertRules - list of reference rules which should be converted
+	convertRules = []string{
+		"ExecutableDefinitionsRule",
+		"FieldsOnCorrectTypeRule",
+		"FragmentsOnCompositeTypesRule",
+		"KnownArgumentNamesRule",
+		"KnownDirectivesRule",
+		"KnownFragmentNamesRule",
+		"KnownTypeNamesRule",
+		"LoneAnonymousOperationRule",
+		"LoneSchemaDefinitionRule",
+		"NoFragmentCyclesRule",
+		"NoUndefinedVariablesRule",
+		"NoUnusedFragmentsRule",
+		"NoUnusedVariablesRule",
+		"OverlappingFieldsCanBeMergedRule",
+		"PossibleFragmentSpreadsRule",
+		"PossibleTypeExtensionsRule",
+		"ProvidedRequiredArgumentsRule",
+		"ScalarLeafsRule",
+		"SingleFieldSubscriptionsRule",
+		"UniqueArgumentNamesRule",
+		"UniqueDirectiveNamesRule",
+		"UniqueDirectivesPerLocationRule",
+		"UniqueEnumValueNamesRule",
+		"UniqueFieldDefinitionNamesRule",
+		"UniqueFragmentNamesRule",
+		"UniqueInputFieldNamesRule",
+		"UniqueOperationNamesRule",
+		"UniqueOperationTypesRule",
+		"UniqueTypeNamesRule",
+		"UniqueVariableNamesRule",
+		"ValuesOfCorrectTypeRule",
+		"VariablesAreInputTypesRule",
+		"VariablesInAllowedPositionRule",
+		// "validation", // should be rewritten manually
+		// "NoDeprecatedCustomRule", // should be ignored we have no custom rules
+		// "NoSchemaIntrospectionCustomRule", // should be ignored we have no custom rules
+	}
+)
+
+// skipRule - determine do we have to convert particular rule e.g. test file from reference tests
+func skipRule(name string) bool {
```

Comment:
if you make `convertRules` as a `map[string]bool` you can remove this function and just use it conveniently as:
```
_, skip := convertRules[rule]
```